src/selectors/entities/channel_categories.ts

```diff
@@ -384,18 +454,20 @@ export function makeGetChannelsForCategory() {
 // with inactive DMs/GMs and archived channels filtered out.
 export function makeFilterAndSortChannelsForCategory() {
     const filterArchivedChannels = makeFilterArchivedChannels();
-
+    const legacyFilterAutoclosedDMs = legacyMakeFilterAutoclosedDMs();
     const filterAutoclosedDMs = makeFilterAutoclosedDMs();
+
     const filterManuallyClosedDMs = makeFilterManuallyClosedDMs();
 
     const sortChannels = makeSortChannels();
 
     return (state: GlobalState, originalChannels: Channel[], category: ChannelCategory) => {
+        const sidebarOrganisationPreference = getNewSidebarPreference(state);
         let channels = originalChannels;
 
         channels = filterArchivedChannels(state, channels);
 
-        channels = filterAutoclosedDMs(state, channels, category.type);
+        channels = sidebarOrganisationPreference ? filterAutoclosedDMs(state, channels, category.type) : legacyFilterAutoclosedDMs(state, channels, category.type);
```

Comment:
Double checking, but is this actually used by the DM/GM channel loading code? If not, these selectors aren't used by the old sidebar, so we don't need to keep `legacyFilterAutoclosedDMs` around any more since the old sidebar is controlled by a different selector.